### PR TITLE
Implement per-node metrics capability config

### PR DESCRIPTION
**Pull request by kralicky**
Closes https://github.com/rancher/opni/issues/1289
Fixes https://github.com/rancher/opni/issues/1288
Fixes https://github.com/rancher/opni/issues/1291
Fixes https://github.com/rancher/opni/issues/1292

### CODE REVIEW

File: pkg/opni/commands/metrics.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ func BuildMetricsCmd() *cobra.Command {
 
 	cmd.AddCommand(BuildCortexAdminCmd())
 	cmd.AddCommand(BuildCortexOpsCmd())
+	cmd.AddCommand(BuildMetricsConfigCmd())
```
Comment (joshmeranda) [inline pkg/opni/commands/metrics.go:17]: would it make more sense to have capabilty configuration commands under the `opni capabilty` command? Feels out of scope

Comment (kralicky) [reply pkg/opni/commands/metrics.go:17]: Plugin-specific commands are grouped together so that they can be compiled out using build tags. The current structure keeps the core commands separated. I agree it's not the most obvious structure. Refactoring would be out of scope for this PR but we can definitely take a look at updating this

Comment (joshmeranda) [reply pkg/opni/commands/metrics.go:17]: Ok works for me

**Review by joshmeranda — DISMISSED** [review]

File: test/plugins/metrics/node_config_test.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	var getConfig = func(agentId string) (*node.MetricsCapabilitySpec, bool, error) {
+		var trailer metadata.MD
+		spec, err := nodeClient.GetNodeConfiguration(context.Background(), &v1.Reference{Id: agentId}, grpc.Trailer(&trailer))
+		if err != nil {
+			return nil, false, err
+		}
+		var isDefault bool
+		if len(trailer["is-default-config"]) > 0 {
+			isDefault = trailer["is-default-config"][0] == "true"
+		}
```
Comment (alexandreLamarre) [inline test/plugins/metrics/node_config_test.go:64]: might be worth putting this in a package since this default check is also used in the CLI

File: pkg/apis/capability/v1/validation.go [hunk 3]
```diff
@@ -15,3 +15,13 @@ func (req *InstallRequest) Validate() error {
 	}
 	return nil
 }
+
+func (req *SyncRequest) Validate() error {
+	if req.Cluster != nil && req.Cluster.Id != "" {
+		// empty string indicates "all" here
```
Comment (alexandreLamarre) [inline pkg/apis/capability/v1/validation.go:21]: should add this comment to the SyncRequest message definition, for IDE completion

**Review by alexandreLamarre — DISMISSED** [review]

**Review by alexandreLamarre — DISMISSED** [review]
